controllers/remote/cluster_cache.go

```diff
@@ -247,6 +273,18 @@ func (t *ClusterCacheTracker) Watch(ctx context.Context, input WatchInput) error
 		return nil
 	}
 
+	_, ok := input.Kind.(*corev1.Node)
+	if ok {
+		a.cache.Stop()
```

Comment:
Is this required?

Reply:
without it, test was complaining about informer already starter. But It's still failing, I need to dig into it a bit.

Reply:
@vincepri I changed the approach slightly. I added a field `Indexes` to `WatchInput` that is passed through `getClusterAccessorLH`.
So now is up to the outer function e.g `watchNodes` to define their opinionated indexes.

Reply:
heh this makes `TestWatches` fail because `WatchClusterNodes` is gated by `ControlPlaneInitializedCondition`, which depends on the control machine having a nodeRef, so when `getNode` is called for that Machine the the watcher is still not running nor the index initialised so it fails with `"Failed to retrieve Node by ProviderID" "error"="Index with name field:spec.providerID does not exist`

Reply:
@vincepri Given the above I'm starting to think that getting back to set the index in `newClusterAccessor` unconditionally might be the most reasonable thing to do. Thoughts?

Reply:
Wouldn't `TestWatches` create a new remote cache with an index?

Reply:
the way it is right now it fails because the code is broken because getNodes tries to use and index which doesn't exist in the client it uses https://github.com/kubernetes-sigs/cluster-api/blob/1f90fa0dc94a66e232cd6a67af01ad3a23018e6b/controllers/machine_controller_noderef.go#L56 so we'd need to let `GetClient` to receive a list on indexes.